include/platform/spi-slave.h

```diff
@@ -0,0 +1,171 @@
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file spi-slave.h
+ * @brief
+ *   This file includes the platform abstraction for SPI slave communication.
+ */
+
+#ifndef SPI_SLAVE_H_
+#define SPI_SLAVE_H_
+
+#include <stdint.h>
+
+#include <openthread-types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup spi-slave SPI Slave
+ * @ingroup platform
+ *
+ * @brief
+ *   This module includes the platform abstraction for SPI slave communication.
+ *
+ * @{
+ *
+ */
+
+/**
+ * Indicates that a SPI transaction has completed with
+ * the given length. The data written to the slave has been
+ * written to the pointer indicated by the `anInputBuf` argument
+ * to the previous call to otPlatSpiSlavePrepareTransaction().
+ *
+ * Once this function is called, otPlatSpiSlavePrepareTransaction()
+ * is invalid and must be called again for the next transaction to be
+ * valid.
+ *
+ * Note that this function is always called at the end of a transaction,
+ * even if otPlatSpiSlavePrepareTransaction() has not yet been called.
+ *
+ * @param[in] aContext Context pointer passed into otPlatSpiSlaveEnable()
+ * @param[in] aLen     Length of the completed transaction, in bytes
+ */
+typedef void (*otPlatSpiSlaveTransactionCompleteCallback)(void *aContext, uint16_t aLen);
+
+/**
+ * Initialize the SPI slave interface.
+
+ * Note that the SPI slave is not fully ready until a transaction is
+ * prepared using otPlatSPISlavePrepareTransaction().
+ *
+ * If otPlatSPISlavePrepareTransaction() is not called before
+ * the master begins a transaction, the resulting SPI transaction
+ * will send all `0xFF` bytes and discard all received bytes.
+ *
+ * @param[in] aCallback Pointer to transaction complete callback
+ * @param[in] aContext  Context pointer to be passed to transaction
+ *                      complete callback
+ *
+ * @retval ::kThreadError_None    Successfully enabled the SPI Slave interface.
+ * @retval ::kThreadError_Already SPI Slave interface is already enabled.
+ * @retval ::kThreadError_Fail    Failed to enable the SPI Slave interface.
+ */
+ThreadError otPlatSpiSlaveEnable(otPlatSpiSlaveTransactionCompleteCallback aCallback, void *aContext);
+
+/**
+ * Shutdown and disable the SPI slave interface.
+ */
+void otPlatSpiSlaveDisable(void);
+
+/**
+ * Prepare data for the next SPI transaction. Data pointers
+ * MUST remain valid until the transaction complete callback
+ * is called by the SPI slave driver, or until after the
+ * next call to otPlatSpiSlavePrepareTransaction().
+ *
+ * This function may be called more than once before the SPI
+ * master initiates the transaction. Each *successful* call to this
+ * function will cause the previous values from earlier calls to
+ * be discarded.
```

Comment:
We discard the earlier values because we have better ones now. There will only ever be a single TX request in flight for this API at a time. This API should not queue.